### PR TITLE
plugin: add battery plugin

### PR DESCRIPTION
**Pull request by sileht**
this change adds the battery plugin

### CODE REVIEW

File: manifests/plugin/battery.pp [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# https://collectd.org/wiki/index.php/Plugin:Uptime
+class collectd::plugin::battery (
+  $ensure   = 'present',
```
Comment (bastelfreak) [inline manifests/plugin/battery.pp]: please add a datatype for `$ensure` as well.

File: spec/classes/collectd_plugin_battery_spec.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+# Generated by Puppet
+<LoadPlugin battery>
+  Globals false
+</LoadPlugin>
+
+<Plugin "battery">
+</Plugin>
+
+EOS
+
```
Comment (bastelfreak) [inline spec/classes/collectd_plugin_battery_spec.rb]: please also add a `is_expected.to compile.with_all_deps` test

File: templates/plugin/battery.conf.erb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<Plugin "battery">
```
Comment (bastelfreak) [inline templates/plugin/battery.conf.erb]: for new templates we agreed on epp only: https://voxpupuli.org/docs/#reviewing-a-module-pr
Can you please convert this to epp?

File: manifests/plugin/battery.pp [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  $interval = undef,
+  Optional[Boolean] $values_percentage = undef,
```
Comment (bastelfreak) [inline manifests/plugin/battery.pp]: can we instead default to true or false and don't mark it as optional?

Comment (sileht) [reply manifests/plugin/battery.pp]: done

File: templates/plugin/battery.conf.erb (hunk 3 above)
Comment (sileht) [reply templates/plugin/battery.conf.erb]: done

File: manifests/plugin/battery.pp (hunk 1 above)
Comment (sileht) [reply manifests/plugin/battery.pp]: done

File: spec/classes/collectd_plugin_battery_spec.rb (hunk 2 above)
Comment (sileht) [reply spec/classes/collectd_plugin_battery_spec.rb]: done

**Pull request merged**